sdk/cosmos/src/requests/query_documents_builder.rs

```diff
@@ -70,10 +58,11 @@ where
     }
 }
 
-impl<'a, 'b> QueryDocumentsBuilder<'a, 'b, Yes> {
-    pub async fn execute<T>(&self) -> Result<QueryDocumentsResponse<T>, CosmosError>
+impl<'a, 'b> QueryDocumentsBuilder<'a, 'b> {
+    pub async fn execute<T, Q>(&self, query: Q) -> Result<QueryDocumentsResponse<T>, CosmosError>
```

Comment:
Could this just be changed to be `impl Into<Query<'a>>` to avoid the second generic you were worried about?

Reply:
No unfortunately mixing `impl Trait` and generic traits doesn't work.

Reply:
Oh right...I forgot this is using generic traits